src/Juvix/Backends/LLVM/Codegen/Block.hs

```diff
@@ -273,6 +275,58 @@ terminator trm = do
   modifyBlock (blk {term = Just trm})
   return trm
 
+--------------------------------------------------------------------------------
+-- External linking
+--------------------------------------------------------------------------------
+
+external ∷ (HasState "moduleDefinitions" [Definition] m) ⇒ Type → String → [(Type, Name)] → m Operand
+external retty label argtys = do
+  addDefn
+    $ GlobalDefinition
+    $ functionDefaults
+      { Global.parameters = ((\(ty, nm) → Parameter ty nm []) <$> argtys, False),
+        Global.callingConvention = CC.Fast, -- TODO: Do we always want this?
+        Global.returnType = retty,
+        Global.basicBlocks = [],
+        Global.name = mkName label,
+        Global.linkage = L.External
+      }
+  return
+    $ ConstantOperand
+    $ C.GlobalReference
+      (PointerType (FunctionType retty (fst <$> argtys) False) (AddrSpace 0))
+      (mkName label)
+
+--------------------------------------------------------------------------------
+-- Memory management
+--------------------------------------------------------------------------------
+
+-- malloc & free need to be defined once and then can be called normally with `externf`
+
+defineMalloc ∷
+  ∀ m.
+  ( HasState "moduleDefinitions" [Definition] m,
+    HasState "symtab" (HashMap Symbol Operand) m
+  ) ⇒
+  m ()
+defineMalloc = do
+  let name = "malloc"
+  op ← external voidStarTy name [(size_t, "size")]
```

Comment:
How much can we change the size of the pointer?

Reply:
I think it depends on the library we're linking against (that provides `malloc` / `free`)

Reply:
(for now we can just worry about 64-bit systems)